cms/src/main/java/uk/nhs/digital/apispecs/module/ApiSpecSyncFromApigeeModule.java

```diff
@@ -28,8 +32,26 @@
     private final Object configurationLock = new Object();
 
     private static final String JOB_GROUP = "devzone";
-    private static final String JOB_NAME = "apiSpecSyncFromApigee";
     private static final String JOB_TRIGGER_NAME = "cronTrigger";
+    private static final String DAILY_JOB_NAME = "apiSpecSyncFromApigee";
+    private static final String NIGHTLY_JOB_NAME = "apiSpecRefresh";
+    private static final String[] JOB_NAMES = {DAILY_JOB_NAME, NIGHTLY_JOB_NAME};
```

Comment:
I see the purpose behind `JOB_NAMES`, `CRON_INDEX` and `CLASS_INDEX` but, I think, we could make it a bit simpler.

Each of these structures (including the `ConfigPropertyNames` enum) contains a collection of job-specific properties, but, at the moment, these are scattered across those disparate collections. By aggregating these properties in one 'collection' keyed on a job type we could make this a bit more object-oriented and easier to follow.

What I'd do is re-define the enum as something like:
```java
    enum JOBS {

        DAILY(
            "apiSpecSyncFromApigee",
            ApiSpecSyncFromApigeeJob.class,
            "devzone.apispec.sync.daily-cron-expression",
            "cronExpression"
        ),
        NIGHTLY(
            "apiSpecRefresh",
            ApiSpecRerenderJob.class,
            "devzone.apispec.sync.nightly-cron-expression",
            "cronExpression"
        );

        private final String systemPropertyName;
        private final String jcrPropertyName;
        private final String jobName;
        private final Class<?> jobClass;

        JOBS(final String jobName, final Class<?> jobClass, final String systemPropertyName, final String jcrPropertyName) {
            this.systemPropertyName = systemPropertyName;
            this.jcrPropertyName = jcrPropertyName;
            this.jobName = jobName;
            this.jobClass = jobClass;
        }
    }
```
...where `doConfigure` could then be turned into something like this:
```java
@Override protected void doConfigure(final Node moduleConfigNode) {

        // ...

        synchronized (configurationLock) {
            Arrays.stream(JOBS.values()).forEach(job -> {
                try {
                    log.info("Configuring job {}/{} for updating API Specifications from Apigee.", JOB_GROUP, job.jobName); // the only one called on the config node change

                    deactivatePreviousJobInstanceIfExist(job);

                    if (jobShouldBeEnabled(moduleConfigNode, job)) {

                        scheduleNewJob(moduleConfigNode, job);

                    } else {
                        log.warn("Job {}/{} is not configured to be enabled, therefore it will not be activated.", JOB_GROUP, job.jobName);
                    }

                } catch (final Exception e) {
                    throw new RuntimeException(String.format("Failed to configure job %s/%s", JOB_GROUP, job.jobName), e);
                }
            });
        }
    }
```
...updating the methods called from it to accept single `job` enum value and retrieve job's properties from it.

Reply:
love this - i wasn't happy with the use of two Maps. hadn't considered expanding the enum but makes perfect sense :+1: